cmd/soroban-cli/src/commands/tx/edit.rs

```diff
@@ -0,0 +1,184 @@
+use std::{
+    env,
+    fs::{self, File},
+    io::{stdin, Cursor, IsTerminal, Write},
+    path::PathBuf,
+    process::{self},
+};
+
+#[cfg(windows)]
+use std::process::Stdio;
+
+use base64::{engine::general_purpose::STANDARD as BASE64, Engine as _};
+use stellar_xdr::curr;
+
+use crate::{commands::global, print::Print};
+
+const DEFAULT_JSON: &str = r#"{
+  "tx": {
+    "tx": {
+      "source_account": "",
+      "fee": 100,
+      "seq_num": 0,
+      "cond": "none",
+      "memo": "none",
+      "operations": [],
+      "ext": "v0"
+    },
+    "signatures": []
+  }
+}
+"#;
+
+#[derive(thiserror::Error, Debug)]
+pub enum Error {
+    #[error(transparent)]
+    Io(#[from] std::io::Error),
+
+    #[error(transparent)]
+    StellarXdr(#[from] stellar_xdr::curr::Error),
+
+    #[error(transparent)]
+    SerdeJson(#[from] serde_json::Error),
+
+    #[error(transparent)]
+    Base64Decode(#[from] base64::DecodeError),
+
+    #[error("Editor returned non-zero status")]
+    EditorNonZeroStatus,
+}
+
+// Command to edit the transaction
+/// e.g. `stellar tx new manage-data --data-name hello --build-only | stellar tx edit`
+#[derive(Debug, clap::Parser, Clone, Default)]
+#[group(skip)]
+pub struct Cmd {}
+
+impl Cmd {
+    pub fn run(&self, global_args: &global::Args) -> Result<(), Error> {
+        let print = Print::new(global_args.quiet);
+        let json: String = if stdin().is_terminal() {
+            DEFAULT_JSON.to_string()
+        } else {
+            let mut input = String::new();
+            stdin().read_line(&mut input)?;
+            let input = input.trim();
+            xdr_to_json::<curr::TransactionEnvelope>(input)?
+        };
+
+        let path = tmp_file(&json)?;
+        let editor = get_editor();
+
+        open_editor(&print, &editor, &path)?;
+
+        let contents = fs::read_to_string(&path)?;
+        let xdr = json_to_xdr::<curr::TransactionEnvelope>(&contents)?;
+        fs::remove_file(&path)?;
+
+        println!("{xdr}");
+
+        Ok(())
+    }
+}
+
+struct Editor {
+    cmd: String,
+    source: String,
+    args: Vec<String>,
+}
+
+fn tmp_file(contents: &str) -> Result<PathBuf, Error> {
+    let temp_dir = env::current_dir().unwrap_or(env::temp_dir());
+    let file_name = format!("stellar-json-xdr-{}.json", rand::random::<u64>());
```

Comment:
Suggest to drop the duplicative 'json', as it's already in the file extension, and we could add the type for context in editors showing the name:
```suggestion
    let file_name = format!("stellar-{}-{}.xdr.json", type_name, rand::random::<u64>());
```

Reply:
what do you mean by "type"? The xdr type?

Reply:
I meant the xdr type.

Reply:
does it make sense in this case, given he xdr will always be a transaction envelope?

Reply:
Oh right, good point, yeah doesn't make sense.

Reply:
I wonder if a generic version of this command could also live at `stellar xdr edit` 🤔

Reply:
Let's wait and see if ask for it. Porting those functions to be a generic thing would be easy enough!